Build failure: fooyin

### Nixpkgs version

- Stable (25.11)

### Steps to reproduce

1. Update to NixOS 25.11 and rebuild the system (coming from 25.05 and fooyin 0.8.1, which was building fine on my end):

```nix
environment.systemPackages =
  with pkgs;
  [
    fooyin
  ];
```

2. `fooyin` fails to build (see full log below).

### Can Hydra reproduce this build failure?

Hydra doesn’t try to build the package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Running phase: qtPreHook
@nix { "action": "setPhase", "phase": "qtPreHook" }
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/3czk84yan21j8mm8846pylqh6wri8c7r-source
source root is source
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/k0jcbkws1xr48rf043hlf47bikp1252i-add-missing-header.patch
patching file src/plugins/vumeter/vumeterwidget.cpp
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/pxqzrjqpviq0jfdkmplpl67gklg0glck-fooyin-0.9.2/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/>
-- The CXX compiler identification is GNU 14.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/vr15iyyykg9zai6fpgvhcgyw7gckl78w-gcc-wrapper-14.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Performing Test HAVE_STDATOMIC
-- Performing Test HAVE_STDATOMIC - Success
-- Found WrapAtomic: TRUE
-- Found OpenGL: /nix/store/mm4v6w3wj8wi5qc6rwdrb2rkqs08whp9-libglvnd-1.7.0/lib/libOpenGL.so
-- Found WrapOpenGL: TRUE
-- Found WrapVulkanHeaders: /nix/store/j0jc9819vx3gdky3jfxp9fwza54xk184-vulkan-headers-1.4.328.0/include
-- Found ICU: /nix/store/xsa61plh8lg41pdpp8hbd143hpczpfh2-icu4c-76.1-dev/include (found version "76.1") found components: uc i18n
-- Found Taglib: /nix/store/b8zlgsxvibjykjkya9dln2zhkj9kndbj-taglib-2.1.1/lib/libtag.so (found version "2.1.1")
-- Found PkgConfig: /nix/store/8vdiwpbh0g4avsd6x5v4s0di32vcl3dp-pkg-config-wrapper-0.29.2/bin/pkg-config (found version "0.29.2")
-- Checking for module 'libavcodec'
--   Found libavcodec, version 62.11.100
-- Checking for module 'libavformat'
--   Found libavformat, version 62.3.100
-- Checking for module 'libavdevice'
--   Found libavdevice, version 62.1.100
-- Checking for module 'libavutil'
--   Found libavutil, version 60.8.100
-- Checking for module 'libavfilter'
--   Found libavfilter, version 11.4.100
-- Checking for module 'libswscale'
--   Found libswscale, version 9.1.100
-- Checking for module 'libswresample'
--   Found libswresample, version 6.1.100
-- Checking for module 'libpostproc'
--   No package 'libpostproc' found
-- Found ZLIB: /nix/store/l7xwm1f6f3zj2x8jwdbi8gdyfbx07sh7-zlib-1.3.1/lib/libz.so (found version "1.3.1")
-- Found FFmpeg: /nix/store/smns0ny2jjfs4vm2y4zv0bdanwx00fdy-ffmpeg-8.0-lib/lib/libavcodec.so;/nix/store/smns0ny2jjfs4vm2y4zv0bdanwx00fdy-ffmpeg-8.0-lib/lib/libavfilter.so;/nix/store/smns0ny2jjfs4vm2y4zv0bdanwx00fdy-ffmpeg-8.0-lib/lib/libavformat.so;/nix/store/smns0ny2jjfs4vm2y4zv0bdanwx00fdy-ffmpeg-8.0-lib/lib/li>
-- Using system KDSingleApplication
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
Package libmpg123 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libmpg123.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libmpg123', required by 'libopenmpt', not found
Package libmpg123 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libmpg123.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libmpg123', required by 'libopenmpt', not found
Package libmpg123 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libmpg123.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libmpg123', required by 'libopenmpt', not found
Package libmpg123 was not found in the pkg-config search path.
Perhaps you should add the directory containing `libmpg123.pc'
to the PKG_CONFIG_PATH environment variable
Package 'libmpg123', required by 'libopenmpt', not found
Package flac was not found in the pkg-config search path.
Perhaps you should add the directory containing `flac.pc'
to the PKG_CONFIG_PATH environment variable
Package 'flac', required by 'sndfile', not found
Package flac was not found in the pkg-config search path.
Perhaps you should add the directory containing `flac.pc'
to the PKG_CONFIG_PATH environment variable
Package 'flac', required by 'sndfile', not found
Package flac was not found in the pkg-config search path.
Perhaps you should add the directory containing `flac.pc'
to the PKG_CONFIG_PATH environment variable
Package 'flac', required by 'sndfile', not found
Package flac was not found in the pkg-config search path.
Perhaps you should add the directory containing `flac.pc'
to the PKG_CONFIG_PATH environment variable
Package 'flac', required by 'sndfile', not found
-- Using system libvgm
--
-- ******** Fooyin Configuration ********
-- General:
--   Fooyin version        : 0.9.2
--   CMake version         : 4.1.2
--   CMake command         : /nix/store/w9jm660dykns6hzrdhxmqfywnc9ail8g-cmake-4.1.2/bin/cmake
--   System                : Linux
--   C++ compiler          : /nix/store/vr15iyyykg9zai6fpgvhcgyw7gckl78w-gcc-wrapper-14.3.0/bin/g++
--   C++ compiler version  : 14.3.0
--   CXX flags             :
--   Build type            : Release
--
-- Install location:
--   Install prefix        : /nix/store/pxqzrjqpviq0jfdkmplpl67gklg0glck-fooyin-0.9.2
--   Binary install dir    : bin
--   Library install dir   : lib/fooyin
--   Plugin install dir    : lib/fooyin/plugins
--
-- Options:
--   BUILD_SHARED_LIBS     : ON
--   BUILD_TESTING         : FALSE
--   BUILD_PLUGINS         : ON
--   BUILD_ALSA            : ON
--   BUILD_LIBVGM          : ON
--   BUILD_TRANSLATIONS    : ON
--   BUILD_CCACHE          : ON
--   BUILD_PCH             : OFF
--   BUILD_WERROR          : OFF
--   BUILD_ASAN            : OFF
--   INSTALL_FHS           : TRUE
--   INSTALL_HEADERS       : OFF
-- Configuring done (8.4s)
-- Generating done (0.4s)
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_C_COMPILER
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
cmake: enabled parallel installing
Running phase: buildPhase
@nix { "action": "setPhase", "phase": "buildPhase" }
build flags: -j8 SHELL=/nix/store/rlq03x4cwf8zn73hxaxnx0zn5q9kifls-bash-5.3p3/bin/bash
[  0%] Built target fooyin_version_autogen_timestamp_deps
[  0%] Built target fooyin_utils_autogen_timestamp_deps
[  1%] Generating data/fooyin_de.qm
[  1%] Generating data/fooyin_en_GB.qm
[  1%] Generating data/fooyin_en.qm
[  1%] Generating data/fooyin_en_US.qm
[  1%] Generating data/fooyin_af.qm
Updating '/build/source/build/data/fooyin_en_GB.qm'...
    Generated 1 translation(s) (1 finished and 0 unfinished)
[  1%] Generating data/fooyin_ca.qm
[  1%] Built target fooyin_config_autogen_timestamp_deps
[  1%] Automatic MOC for target fooyin_version
Updating '/build/source/build/data/fooyin_en_US.qm'...
[  1%] Generating data/fooyin_es.qm
    Generated 1544 translation(s) (1529 finished and 15 unfinished)
    Ignored 30 untranslated source text(s)
[  2%] Automatic MOC for target fooyin_utils
Updating '/build/source/build/data/fooyin_en.qm'...
    Generated 0 translation(s) (0 finished and 0 unfinished)
    Ignored 1574 untranslated source text(s)
[  3%] Generating data/fooyin_fi.qm
Updating '/build/source/build/data/fooyin_es.qm'...
Updating '/build/source/build/data/fooyin_af.qm'...
    Generated 110 translation(s) (29 finished and 81 unfinished)
    Ignored 1464 untranslated source text(s)
    Generated 749 translation(s) (454 finished and 295 unfinished)
    Ignored 825 untranslated source text(s)
[  4%] Automatic MOC for target fooyin_config
Updating '/build/source/build/data/fooyin_de.qm'...
    Generated 886 translation(s) (600 finished and 286 unfinished)
    Ignored 688 untranslated source text(s)
[  4%] Generating data/fooyin_id.qm
[  4%] Generating data/fooyin_fr.qm
Updating '/build/source/build/data/fooyin_ca.qm'...
Updating '/build/source/build/data/fooyin_fi.qm'...
    Generated 243 translation(s) (86 finished and 157 unfinished)
    Ignored 1331 untranslated source text(s)
[  4%] Generating data/fooyin_ie.qm
    Generated 1546 translation(s) (1494 finished and 52 unfinished)
    Ignored 28 untranslated source text(s)
[  4%] Generating data/fooyin_it.qm
[  5%] Generating data/fooyin_ja.qm
Updating '/build/source/build/data/fooyin_fr.qm'...
Updating '/build/source/build/data/fooyin_id.qm'...
    Generated 1545 translation(s) (1519 finished and 26 unfinished)
    Ignored 29 untranslated source text(s)
    Generated 202 translation(s) (64 finished and 138 unfinished)
    Ignored 1372 untranslated source text(s)
Updating '/build/source/build/data/fooyin_ie.qm'...
    Generated 452 translation(s) (286 finished and 166 unfinished)
    Ignored 1122 untranslated source text(s)
[  5%] Generating data/fooyin_lt.qm
Updating '/build/source/build/data/fooyin_it.qm'...
    Generated 198 translation(s) (69 finished and 129 unfinished)
    Ignored 1376 untranslated source text(s)
[  5%] Generating data/fooyin_nb_NO.qm
[  5%] Built target fooyin_version_autogen
[  5%] Generating data/fooyin_nl.qm
[  5%] Generating data/fooyin_pl.qm
Updating '/build/source/build/data/fooyin_ja.qm'...
    Generated 469 translation(s) (236 finished and 233 unfinished)
    Ignored 1105 untranslated source text(s)
[  5%] Built target fooyin_config_autogen
Updating '/build/source/build/data/fooyin_lt.qm'...
    Generated 683 translation(s) (405 finished and 278 unfinished)
    Ignored 891 untranslated source text(s)
[  5%] Generating data/fooyin_pt.qm
Updating '/build/source/build/data/fooyin_nl.qm'...
Updating '/build/source/build/data/fooyin_pl.qm'...
[  6%] Generating data/fooyin_pt_BR.qm
    Generated 775 translation(s) (520 finished and 255 unfinished)
    Ignored 799 untranslated source text(s)
[  6%] Building CXX object CMakeFiles/fooyin_config.dir/fooyin_config_autogen/mocs_compilation.cpp.o
    Generated 780 translation(s) (403 finished and 377 unfinished)
    Ignored 794 untranslated source text(s)
Updating '/build/source/build/data/fooyin_nb_NO.qm'...
    Generated 344 translation(s) (140 finished and 204 unfinished)
    Ignored 1230 untranslated source text(s)
[  6%] Generating data/fooyin_sk.qm
[  6%] Generating data/fooyin_ru.qm
[  6%] Building CXX object CMakeFiles/fooyin_version.dir/fooyin_version_autogen/mocs_compilation.cpp.o
[  6%] Generating data/fooyin_sv.qm
Updating '/build/source/build/data/fooyin_pt.qm'...
    Generated 795 translation(s) (596 finished and 199 unfinished)
    Ignored 779 untranslated source text(s)
[  6%] Generating data/fooyin_th.qm
Updating '/build/source/build/data/fooyin_pt_BR.qm'...
    Generated 1128 translation(s) (1072 finished and 56 unfinished)
    Ignored 446 untranslated source text(s)
[  7%] Generating data/fooyin_tr.qm
[  7%] Built target fooyin_version
Updating '/build/source/build/data/fooyin_ru.qm'...
[  7%] Built target fooyin_config
    Generated 1545 translation(s) (1533 finished and 12 unfinished)
    Ignored 29 untranslated source text(s)
Updating '/build/source/build/data/fooyin_th.qm'...
    Generated 149 translation(s) (37 finished and 112 unfinished)
    Ignored 1425 untranslated source text(s)
[  7%] Generating data/fooyin_uk.qm
Updating '/build/source/build/data/fooyin_sk.qm'...
    Generated 386 translation(s) (184 finished and 202 unfinished)
    Ignored 1188 untranslated source text(s)
Updating '/build/source/build/data/fooyin_sv.qm'...
[  7%] Generating data/fooyin_zh_CN.qm
[  7%] Generating data/fooyin_zh_Hant.qm
    Generated 431 translation(s) (213 finished and 218 unfinished)
    Ignored 1143 untranslated source text(s)
Updating '/build/source/build/data/fooyin_tr.qm'...
    Generated 435 translation(s) (199 finished and 236 unfinished)
    Ignored 1139 untranslated source text(s)
Updating '/build/source/build/data/fooyin_uk.qm'...
    Generated 1544 translation(s) (1533 finished and 11 unfinished)
    Ignored 30 untranslated source text(s)
Updating '/build/source/build/data/fooyin_zh_CN.qm'...
    Generated 1542 translation(s) (1528 finished and 14 unfinished)
    Ignored 32 untranslated source text(s)
Updating '/build/source/build/data/fooyin_zh_Hant.qm'...
    Generated 1545 translation(s) (1533 finished and 12 unfinished)
    Ignored 29 untranslated source text(s)
[  7%] Built target fooyin_lrelease
[  7%] Built target fooyin_utils_autogen
[  8%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/itemregistry.cpp.o
[  8%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/audioutils.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/fooyin_utils_autogen/mocs_compilation.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/fileutils.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/crypto.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/id.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/datastream.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/modelutils.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/paths.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/signalthrottler.cpp.o
[  9%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/stareditor.cpp.o
[ 10%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/stardelegate.cpp.o
[ 10%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/starrating.cpp.o
[ 10%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/stringcollator.cpp.o
[ 10%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/stringutils.cpp.o
[ 10%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/timer.cpp.o
[ 11%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/tooltipfilter.cpp.o
[ 11%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/treemodel.cpp.o
/build/source/src/utils/starrating.cpp: In member function ‘void Fooyin::StarRating::paint(QPainter*, const QRect&, const QPalette&, EditMode, Qt::Alignment) const’:
/build/source/src/utils/starrating.cpp:107:38: error: no matching function for call to ‘QString::arg(Qt::Alignment&)’
  102 |     const QString cacheKey = u"StarRating:%1|%2|%3|%4|%5|%6"_s.arg(m_rating)
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  103 |                                  .arg(m_scale)
      |                                  ~~~~~~~~~~~~~
  104 |                                  .arg(m_maxCount)
      |                                  ~~~~~~~~~~~~~~~~
  105 |                                  .arg(mode == EditMode::Editable ? 1 : 0)
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  106 |                                  .arg(rect.width())
      |                                  ~~~~~~~~~~~~~~~~~~
  107 |                                  .arg(alignment);
      |                                  ~~~~^~~~~~~~~~~
In file included from /nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qhashfunctions.h:9,
                 from /nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qfloat16.h:10,
                 from /nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qmetatype.h:15,
                 from /nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/QMetaType:1,
                 from /build/source/include/utils/starrating.h:24,
                 from /build/source/src/utils/starrating.cpp:20:
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:324:27: note: candidate: ‘template<class T, typename std::enable_if<conjunction_v<std::disjunction<std::is_integral<_Tp>, std::conjunction<std::is_enum<_Tp>, std::negation<std::is_scoped_enum<T> > > >, std::negation<std::disjunction>
  324 |     [[nodiscard]] QString arg(T a, int fieldWidth = 0, int base = 10,
      |                           ^~~
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:324:27: note:   template argument deduction/substitution failed:
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:323:53: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  323 |     template <typename T, if_integral_non_char<T> = true>
      |                                                     ^~~~
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:339:27: note: candidate: ‘template<class T, typename std::enable_if<std::disjunction<std::is_same<typename std::remove_cvref<_Tp>::type, _Float16>, std::is_same<typename std::remove_cvref<_Tp>::type, qfloat16>, std::is_floating_poin>
  339 |     [[nodiscard]] QString arg(T a, int fieldWidth = 0, char format = 'g', int precision = -1,
      |                           ^~~
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:339:27: note:   template argument deduction/substitution failed:
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:338:50: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  338 |     template <typename T, if_floating_point<T> = true>
      |                                                  ^~~~
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:344:27: note: candidate: ‘template<class T, typename std::enable_if<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv< <template-parameter-1-1> >::type> >, std::is_convertible<T, QAnyStringView>>
  344 |     [[nodiscard]] QString arg(const T &a, int fieldWidth = 0, QChar fillChar = u' ') const
      |                           ^~~
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:344:27: note:   template argument deduction/substitution failed:
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:343:47: error: no type named ‘type’ in ‘struct std::enable_if<false, bool>’
  343 |     template <typename T, if_string_like<T> = true>
      |                                               ^~~~
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:364:5: note: candidate: ‘template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_conv>
  364 |     arg(Args &&...args) const
      |     ^~~
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:364:5: note:   template argument deduction/substitution failed:
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h: In substitution of ‘template<class ... Args> typename std::enable_if<((sizeof... (Args) >= 2) && conjunction_v<std::conjunction<std::negation<QtPrivate::treat_as_integral_arg<typename std::remove_cv<Args>::type> >, std::is_converti>
/build/source/src/utils/starrating.cpp:107:38:   required from here
  102 |     const QString cacheKey = u"StarRating:%1|%2|%3|%4|%5|%6"_s.arg(m_rating)
      |                              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  103 |                                  .arg(m_scale)
      |                                  ~~~~~~~~~~~~~
  104 |                                  .arg(m_maxCount)
      |                                  ~~~~~~~~~~~~~~~~
  105 |                                  .arg(mode == EditMode::Editable ? 1 : 0)
      |                                  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  106 |                                  .arg(rect.width())
      |                                  ~~~~~~~~~~~~~~~~~~
  107 |                                  .arg(alignment);
      |                                  ~~~~^~~~~~~~~~~
/nix/store/zp6r9bxds1hldvkx1vbrk0d1ady17zhh-qtbase-6.10.1/include/QtCore/qstring.h:364:5: error: no type named ‘type’ in ‘struct std::enable_if<false, QString>’
  364 |     arg(Args &&...args) const
      |     ^~~
[ 11%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/utils.cpp.o
[ 11%] Building CXX object src/utils/CMakeFiles/fooyin_utils.dir/worker.cpp.o
make[2]: *** [src/utils/CMakeFiles/fooyin_utils.dir/build.make:253: src/utils/CMakeFiles/fooyin_utils.dir/starrating.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1135: src/utils/CMakeFiles/fooyin_utils.dir/all] Error 2
make: *** [Makefile:156: all] Error 2
```

### Additional context

Searching for `fooyin` on Hydra just gives me

```
500 Internal Server Error

DBIx::Class::Storage::DBI::_dbh_execute(): DBI Exception: DBD::Pg::st execute failed: ERROR: canceling statement due to statement timeout [for Statement "SELECT me.id, me.finished, me.timestamp, me.jobset_id, me.job, me.nixname, me.description, me.drvpath, me.system, me.license, me.homepage, me.maintainers, me.maxsilent, me.timeout, me.ischannel, me.iscurrent, me.priority, me.globalpriority, me.starttime, me.stoptime, me.iscachedbuild, me.buildstatus, me.size, me.closuresize, me.releasename, me.keep, me.notificationpendingsince FROM builds me LEFT JOIN buildoutputs buildoutputs ON buildoutputs.build = me.id WHERE ( buildoutputs.path ILIKE ? ) ORDER BY id desc LIMIT ?" with ParamValues: 1='%fooyin%', 2='10'] at /nix/store/jsznzvlv929jxams73sjipnxlf9nw54c-hydra-0.1.20251125.3bd8700/libexec/hydra/lib/Hydra/Controller/Root.pm line 545
```

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.59, NixOS, 25.11 (Xantusia), 25.11.20251130.8bb5646`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixos-25.05"`
 - nixpkgs: `/nix/store/dgp9wmspgm6zhl66f0c81rvg3v7wwbsb-source`

### Notify maintainers

@peterhoeg 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/fo/fooyin/package.nix
+++ b/pkgs/by-name/fo/fooyin/package.nix
@@ -79,6 +79,12 @@ stdenv.mkDerivation (finalAttrs: {
     })
   ];
 
+  # Fix compatibility with Qt 6.10.1 - should be fixed in next release
+  postPatch = ''
+    substituteInPlace src/utils/starrating.cpp \
+      --replace-fail '.arg(alignment);' '.arg(alignment.toInt());'
+  '';
+
   meta = {
     description = "Customisable music player";
     homepage = "https://www.fooyin.org/";

